### ISSUE TITLE
sitewide 508-defect-2 [COGNITION, SCREEN READER]: Screen reader users MUST know there is additional information after making a selection in the wizard

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#18581]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh Kim

## User Story or Problem Statement

As a screen reader user, I need to know that there is additional relevant information after completing the date input

## Details

This is a sitewide defect documented in ticket #316 

## WCAG or Vendor Guidance (optional)

* [WCAG 2.1 Level A - 1.3.2 Meaningful Sequence](https://www.wuhcag.com/meaningful-sequence/) - Users who rely on assistive technology (such as a screen reader) to interpret content, require content to be presented in a meaningful order. If this is presented out of sequence, users may become disorientated and will not understand the content.

## Screenshots or Trace Logs
<img width="525" alt="Screen Shot 2021-02-09 at 1 46 30 PM" src="https://user-images.githubusercontent.com/14154792/107412044-3a198600-6add-11eb-9be4-0837de16cf81.png">

**Comment by joshkimux** [comment]
@Mottie [I did some testing with aria-describedby for a similar alert](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19660) on another ticket and noticed that there's a strong possibility users may tab past it. In cases where there's warnings or alerts, and not just additional information, we may want to make a push for using live regions-- especially in cases where the legend is particularly long.

**Comment by joshkimux** [comment]
[Bumping this up to a 508-defect-1 according to our documented guidelines](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1).

> These issues have the potential to severely disrupt the user experience and must be fixed before VSP will certify a staging accessibility review:
> - [Content being added or removed from the page must announce a change to assistive devices](https://fae.disability.illinois.edu/rulesets/WIDGET_14/)

**Comment by Mottie** [comment]
Ok, sounds good. I'll probably chat with you a bit more once I get some time to work on it later today.

**Comment by joshkimux** [comment]
Thanks Mottie!! Looking forward to it!!

**Comment by joshkimux** [comment]
Update: I was going through this again and noticed the alert got bumped down to just standard text. Does anyone have context on the reason for removing the alert status? @kevinstachura @Mottie @luke-gcio 

<img width="525" alt="Screen Shot 2021-02-09 at 1 46 30 PM" src="https://user-images.githubusercontent.com/14154792/107412044-3a198600-6add-11eb-9be4-0837de16cf81.png">

**Comment by kevinstachura** [comment]
Hey @joshkimux! I believe this was part of a more global effort to remove alerts where not necessary. Really testing my memory here but I think this was implemented as part of the mobile experience on wizards. The alert had a lot of unused space and the need for the alert pattern (i.e. the bold text, icon, and color) was incongruent with the message itself. Is your question more around why that first sentence is not bolded?

**Comment by joshkimux** [comment]
Nope, that's perfect! This definitely does help the mobile experience.
If design is on board with this not semantically being considered as an alert, then I can rewrite this ticket!

**Comment by joshkimux** [comment]
I'm bumping the severity down to a defect-level-2 because:
- there is still a link that users can tab to for more information
- it is no longer considered an alert/warning which implies this information is critical to the user

**Comment by joshkimux** [comment]
That being said, we'll need to rethink about how we handle this for the short term and long term. The main issue requiring surgery right now is the wizard and its lack of an explicit continue/submit button.

## Short term (band aid) options
- Set up a live region and announce the additional information using `aria-live="polite"`. We would not use `role="alert"` here since we would not want to hijack the user's experience with `assertive` for additional information.
- Rewrite and reorder the content by placing the link first (first tab stop after the date fields) followed by paragraphs linked to it using `aria-describedby` (see example below)

https://user-images.githubusercontent.com/14154792/107414659-4fdc7a80-6ae0-11eb-9f37-ace87463adb8.mov

----

## Long term (surgery) solution: 
- Replace the wizard with a check eligibility pattern where eligibility questions are provided one at a time with continue buttons followed by an explicit results page

**Comment by Mottie** [comment]
All wizard alerts include an `aria-live="polite"`. I'm not sure if that is the best or final solution, but I'm going to call this ticket done (on my end) until I find out otherwise.

**Comment by joshkimux** [comment]
Tested yesterday and this is working as good as it can given the current nature of the wizard. Thank you @Mottie again for all the hard work!
